include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -411,22 +411,9 @@ SECTIONS
 
 #if defined(CONFIG_APP_SHARED_MEM)
 #define APP_SHARED_ALIGN . = ALIGN(_region_min_align);
-#if defined(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT)
-		#include <app_smem.ld>
-#else
-	SECTION_PROLOGUE(_APP_SMEM_SECTION_NAME, (OPTIONAL),)
-	{
-		APP_SHARED_ALIGN;
-		_app_smem_start = .;
-		APP_SMEM_SECTION()
-		/* Align the end of shared app mem section with the
-		 * minimum granularity required by MPU.
-		 */
-		APP_SHARED_ALIGN;
-		_app_smem_end = .;
+#define SMEM_PARTITION_ALIGN MPU_ALIGN
```

Comment:
oh that's because the generated ld script is common for all architectures. it will be confusing if I mentioned MPU in x86 therefore a macro which is agnostic is being used.